misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JBallerinaDebugServer.java

```diff
@@ -296,6 +297,13 @@ ClientConfigHolder getClientConfigHolder() {
                     .filter(JBallerinaDebugServer::isValidFrame)
                     .toArray(StackFrame[]::new);
 
+            if (context.getLastInstruction() == DebugInstruction.STEP_IN
+                && validFrames[0].getSource().getPath().equals(topStackFrame.getSource().getPath())
+                && validFrames[0].getLine().intValue() == topStackFrame.getLine()) {
+                eventProcessor.sendStepRequest(args.getThreadId(), StepRequest.STEP_INTO);
+                context.setLastInstruction(null);
```

Comment:
Why are we setting `null` as the last instruction?

Reply:
Fixed

Reply:
Here I'm making sure `STEP_IN` request is sent only once, and after the request is sent I'm making the last instruction to null. 

Alternatively, we can maintain a flag to control the number of `STEP_IN` request. WDYT?

Reply:
i"m still not clear how we are receiving multiple step in requests from the client for a single event.

Reply:
We can get multiple `stackTrace` request for a single breakpoint event. Possible `stackTrace` requests are

1.  To represent debug hit navigation bar
2. To represent call stack information.